### PR TITLE
Add rejectResponderTermination prop to TextInput

### PR DESCRIPTION
**Pull request by adamski**
This PR exposes the `rejectResponderTermination` prop to TextInput. This allows components such as [SwipeListView](https://github.com/jemise111/react-native-swipe-list-view) to be swipeable from the TextInput on iOS, as is the case on Android by default. 

See [this article](https://medium.com/@mikeshi/react-native-swipeouts-textinputs-animations-and-gesture-responders-a6baa4745735#.dzqldlap6) for some analysis of the problem. The author of that article created a [copy-and-paste alteration of TextInput](https://gist.github.com/adamski/d16895ded0d4a9e2e7df8594b5e9d3ab), however I wanted a more future proof version which is why I am suggesting the change in TextInput. 

Tested in the iOS simulator, I'm not sure how to provide test examples for this change. 

Thank you!

### CODE REVIEW

**Comment by adamski** [comment]
Can anyone tell me why the Travis test failed?

**Comment by hramos** [comment]
Re-running Travis tests

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by outpunk** [comment]
@mkonicek @hramos any progress with this?

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -492,6 +492,12 @@ const TextInput = React.createClass({
     inlineImagePadding: PropTypes.number,
 
     /**
+     * Allow TextInput to pass touch event to parent.
+     * @platform ios
```
Comment (mkonicek) [inline Libraries/Components/TextInput/TextInput.js]: Why is this iOS only? Looks like a JS change. Does it not work on Android?

File: Libraries/Components/TextInput/TextInput.js [hunk 2]
```diff
@@ -492,6 +492,12 @@ const TextInput = React.createClass({
     inlineImagePadding: PropTypes.number,
 
     /**
+     * Allow TextInput to pass touch event to parent.
```
Comment (mkonicek) [inline Libraries/Components/TextInput/TextInput.js]: I don't understand this comment. Could you extend it please explaining how and why someone would want to use this prop?

**Comment by mkonicek** [comment]
Trying to merge again.

File: Libraries/Components/TextInput/TextInput.js (hunk 1 above)
Comment (adamski) [inline Libraries/Components/TextInput/TextInput.js]: It does not make any difference on Android, where the default behaviour is for a TextInput to pass the touch event to the parent.

File: Libraries/Components/TextInput/TextInput.js (hunk 2 above)
Comment (adamski) [inline Libraries/Components/TextInput/TextInput.js]: OK, will update the comment, thanks

**Comment by adamski** [comment]
Hi guys, any movement on merging this PR?

**Comment by outpunk** [comment]
@mkonicek any news?

**Comment by shergin** [comment]
I think to merge this PR we have to figure out why this prop was set up this way in the first place. And why it made sense only for iOS.
May be we have simple to remove this.

**Comment by shergin** [comment]
cc @ericvicenti 
It was introduced in D1794136 awhile ago. Eric, do you have any idea why this is applicable only for iOS? Do we still have to have it?

**Comment by adamski** [comment]
Not sure what I updates are required, its definitely still relevant though.

**Comment by outpunk** [comment]
@shergin @mkonicek Any news?

**Comment by cmcewen** [comment]
I just cherry-picked and put up a new PR to try and get this merged https://github.com/facebook/react-native/pull/16755

**Comment by hramos** [comment]
Looks like a bad rebase here, but #16755 is a duplicate that should work.

**Pull request closed without merging**